src/rpdk/core/project.py

```diff
@@ -1345,21 +1352,68 @@ def _generate_stack_template_files(self) -> None:
             with ct_file.open("r") as f:
                 json_data = json.load(f)
             resource_name = self.type_info[2]
-            stack_template_data = {
-                "Description": f"Template for {self.type_name}",
-                "Resources": {
-                    f"{resource_name}": {
-                        "Type": self.type_name,
-                        "Properties": self._replace_dynamic_values(
-                            json_data["CreateInputs"]
-                        ),
-                    }
-                },
-            }
-            stack_template_file_name = f"{CANARY_FILE_PREFIX}{count}_001.yaml"
-            stack_template_file_path = stack_template_folder / stack_template_file_name
-            with stack_template_file_path.open("w") as stack_template_file:
-                yaml.dump(stack_template_data, stack_template_file, indent=2)
+
+            self._save_stack_template_data(
+                resource_name,
+                count,
+                stack_template_folder,
+                self._replace_dynamic_values(
+                    json_data[CREATE_INPUTS_KEY],
+                ),
+                CANARY_FILE_CREATE_SUFFIX,
+            )
+            if PATCH_INPUTS_KEY in json_data:
+                supported_patch_inputs = self._translate_supported_patch_inputs(
+                    json_data[PATCH_INPUTS_KEY]
+                )
+                patch_data = jsonpatch.apply_patch(
+                    json_data[CREATE_INPUTS_KEY], supported_patch_inputs, in_place=False
+                )
+                self._save_stack_template_data(
+                    resource_name,
+                    count,
+                    stack_template_folder,
+                    patch_data,
+                    CANARY_FILE_UPDATE_SUFFIX,
+                )
+
+    def _save_stack_template_data(
+        self,
+        resource_name,
+        contract_test_input_count,
+        stack_template_folder,
+        properties_data,
+        suffix,
+    ):
+        stack_template_data = {
+            "Description": f"Template for {self.type_name}",
+            "Resources": {
+                f"{resource_name}": {
+                    "Type": self.type_name,
+                    "Properties": properties_data,
+                }
+            },
+        }
+        stack_template_file_name = (
+            f"{CANARY_FILE_PREFIX}{contract_test_input_count}_{suffix}.yaml"
```

Comment:
This will create files that look like

- For Create
    - AWS::S3::Bucket1001.yaml
    - AWS::S3::Bucket2001.yaml
    - AWS::S3::Bucket3001.yaml
- For Update
    - AWS::S3::Bucket1002.yaml
    - AWS::S3::Bucket2002.yaml
    - AWS::S3::Bucket3002.yaml
    
Is there a reason we didn't instead using a convention more like `{CANARY_FILE_PREFIX}[UPDATE|CREATE]{contract_test_input_count::padding3}.yaml`?

That way names would look like `AWS::S3::BucketUpdateCREATE001`, `AWS::S3::BucketUpdateCREATE002` ...

Reply:
We choose to favor backwards compatibility over changing naming. Since existing canaries all use the `canary1_001`, `canary1_002`, etc. convention, we wanted auto generated canaries to be consistent with the existing naming convention. If we were to change naming conventions, I think it would be advisable to migrate historical canaries to the new naming convention.